doc/investigations/300-joint-ruptures/readme.md

```diff
@@ -0,0 +1,168 @@
+
+Scientific:
+- There does not seem to be a good model of how joint ruptures are formed or what their general shape is.
+- How do we calculate magnitude?
+
+Technical:
+
+- Computational complexity based on the combinatorial explosion of combining subduction and crustal clusters.
+    - Approaches to mitigate this include thinning and combining pre-made ruptures.
+- Large rupture sets, at least during exploratory phase
+    - OOM in opensha
+    - external tools are struggling
+- Joint ruptures most likely have splays. 
+    - Parts of our NZSHM22 tooling explicitly and implicitly assume single-stranded ruptures.
+- Subduction thinning is required, and possibly crustal as well.
+    - We abandoned subduction thinning in NZSHM22.
+- How do we visualise joint ruptures?
+    - For example fishbone diagrams
+    - How to distinguish crustal and subduction sections in ruptoons?
+- What is the length of a joint rupture?
+- How do we validate our ruptures if there's no cohesive scientific model?
+    - possibly RSQSim output
+
+Several ways to generate joint ruptures were investigated which I will describe in more detail in the rest of this document.
+
+## Combine NZSHM22 Ruptures Into Single-Stranded Joint Ruptures
+
+![rupture419779.png](rupture419779.png)
+
+Driver class is [JointRupturePostProcessor](..%2F..%2F..%2Fsrc%2Fmain%2Fjava%2Fnz%2Fcri%2Fgns%2FNZSHM22%2Fopensha%2Fruptures%2Fexperimental%2Fjoint%2FJointRupturePostProcessor.java)
+
+Heavy lifting happens in [JointRuptureBuilder](..%2F..%2F..%2Fsrc%2Fmain%2Fjava%2Fnz%2Fcri%2Fgns%2FNZSHM22%2Fopensha%2Fruptures%2Fexperimental%2FJointRuptureBuilder.java)
+
+Approach: take an existing rupture set that contains both crustal and subduction ruptures and create a rupture set with joint ruptures by combining them.
+
+Very limited in that it only allows jumps to the first and last section in the top row of a subduction rupture. Based on distance to first or last section in crustal fault.
+
+We contributed improvements to `opensha` to enable some of these features, for example to be able to deal with large rupture index files.
+
+## Combine NZSHM22 Ruptures Into Parallel Joint Ruptures
+
+![splays.png](splays.png)
+
+Driver class is in [JointRupturePostProcessor](..%2F..%2F..%2Fsrc%2Fmain%2Fjava%2Fnz%2Fcri%2Fgns%2FNZSHM22%2Fopensha%2Fruptures%2Fexperimental%2Fjoint%2FJointRupturePostProcessor.java)
+
+Heavy lifting happens in [JointRuptureBuilderParallel](..%2F..%2F..%2Fsrc%2Fmain%2Fjava%2Fnz%2Fcri%2Fgns%2FNZSHM22%2Fopensha%2Fruptures%2Fexperimental%2FJointRuptureBuilderParallel.java)
+
+Similar approach to building single-stranded ruptures, but here we combine a subduction and a crustal rupture if a minimum number of sections are close enough. More than one crustal rupture can be linked to a subduction rupture, resulting in more than one splay.
+
+Any other physical features such as rake, dip, relative azimuth are not yet taken into account.
+
+## Use Existing Coulomb Filters to Create Joint Ruptures
```

Comment:
would it be worth a brief summary of how the "standard" Coulomb filter works? That would help put the new Coulomb filter into perspective

Reply:
done